bin/metrics.js

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env node
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+var fs = require('fs')
+var section
+
+module.exports = {
+  run: run,
+  countAccounts: countAccounts,
+  countVerifiedAccounts: countVerifiedAccounts,
+  countAccountsWithTwoOrMoreDevices: countAccountsWithTwoOrMoreDevices,
+  countAccountsWithThreeOrMoreDevices: countAccountsWithThreeOrMoreDevices,
+  countAccountsWithMobileDevice: countAccountsWithMobileDevice
+}
+
+if (require.main === module) {
+  module.exports.run(
+    fs.readFileSync(
+      process.argv[2] || '/etc/gather_basic_metrics.conf',
+      { encoding: 'utf8' }
+    )
+    .split('\n')
+    .map(trim)
+    .filter(filterConfig)
+    .reduce(reduceConfig, {})
+  )
+}
+
+function run (config, now) {
+  now = now || new Date()
+  var lastMidnight = Date.UTC(
+    now.getUTCFullYear(),
+    now.getUTCMonth(),
+    now.getUTCDate(),
+    0, 0, 0, 0
+  )
+  var os = require('os')
+  var log = require('../lib/logging')('bin.metrics')
+  var mysql = require('../lib/db/mysql')(log, require('../fxa-auth-db-server').errors)
+  var P = require('../lib/promise')
+  var self = this
+
+  return mysql.connect({
+    master: {
+      host: config.General.db_dnsname,
+      user: config.General.db_username,
+      password: config.General.db_password,
+      database: config.General.db_name
+    },
+    slave: {
+      host: config.General.db_dnsname,
+      user: config.General.db_username,
+      password: config.General.db_password,
+      database: config.General.db_name
+    },
+    patchKey: 'schema-patch-level'
+  })
+  .then(function (db) {
+    return P.all([
+      self.countAccounts(db, lastMidnight),
+      self.countVerifiedAccounts(db, lastMidnight),
+      self.countAccountsWithTwoOrMoreDevices(db, lastMidnight),
+      self.countAccountsWithThreeOrMoreDevices(db, lastMidnight),
+      self.countAccountsWithMobileDevice(db, lastMidnight)
+    ])
```

Comment:
We probably want to enforce that these run in series, any concurrency between them will be of minimal benefit to the script and will tie up additional server resources.